util/kvencoder/kv_encoder_test.go

```diff
@@ -438,6 +438,35 @@ func (s *testKvEncoderSuite) TestAllocatorRebase(c *C) {
 	sql = "insert into t(id, a) values(2000, 'test')"
 	encoder.Encode(sql, tableID)
 	c.Assert(alloc.Base(), Equals, int64(2000))
+	c.Assert(alloc.End(), Equals, int64(2000)+step)
+	nextID, err := alloc.NextGlobalAutoID(tableID)
+	c.Assert(err, IsNil)
+	c.Assert(nextID, Equals, int64(2000)+step+1)
+}
+
+func (s *testKvEncoderSuite) TestError(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	alloc := NewAllocator()
+	encoder, err := New("test", alloc)
+	c.Assert(err, IsNil)
+	defer encoder.Close()
+	_, err = New("", alloc)
+	c.Assert(err, NotNil)
```

Comment:
I think to use `c.Assert( err, ErrorMatches , regex)` is better than this.
May we could test the error message better.

Reply:
Agree

Reply:
done.